### PR TITLE
Docs for logging multiple val and test dataloaders

### LINKED ISSUE #20885: Logging in `on_test_epoch_end` with multiple dataloaders
### Bug description

Hello,
I am running `test()` with multiple dataloaders and do the logging in `on_test_epoch_end()`. I basically follow https://github.com/Lightning-AI/pytorch-lightning/pull/16520 regarding the "Example with two DataLoaders".

In my case, I log different metrics for the different dataloaders (because each dataloader corresponds to a set set and I directly put the test set name into the logged name). For example:
- dataloader 1 logs `clevr3d_cropped/ari`
- dataloader 2 logs `clevr3d/ari`

However, the final table that is printed by PyTorch-Lightning, shows each logged value duplicated for each of the two dataloaders, which is incorrect:
```
        Test metric               DataLoader 0               DataLoader 1
─────────────────────────────
     clevr3d/ari         0.6213954091072083         0.6213954091072083
 clevr3d_cropped/ari      0.5558694005012512         0.5558694005012512
─────────────────────────────
```

Correctly, the table should look as follows:
```
        Test metric               DataLoader 0               DataLoader 1
─────────────────────────────
     clevr3d/ari         0.6213954091072083                -
 clevr3d_cropped/ari           -                          0.5558694005012512
─────────────────────────────
```

I think the problem is that `self.log()` does not know for which dataloader the log is and therefore logs for both dataloaders. 

I also tried appending the dataloader index in the log, as described in the example mentioned above, but this didn't change anything.

My question now is: is this expected behaviour, or is it a bug, or am I using PyTorch-Lightning incorrectly?

Thanks and best,
Philipp

### What version are you seeing the problem on?

v2.5

### Reproduced in studio

_No response_

### How to reproduce the bug

```python

```

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- PyTorch Lightning Version (e.g., 2.5.0):
#- PyTorch Version (e.g., 2.5):
#- Python version (e.g., 3.12):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
```

</details>

### More info

_No response_

### PR DESCRIPTION
**Pull request by matsumotosan**
## What does this PR do?

Fixes #20885 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21054.org.readthedocs.build/en/21054/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: src/lightning/pytorch/core/module.py [hunk 1]
```diff
@@ -808,7 +808,22 @@ def validation_step(self, batch, batch_idx):
...
+                    loss = self.loss0(out, y)
+                else:
+                    loss = self.loss1(out, y)
+
+                # calculate acc
+                labels_hat = torch.argmax(out, dim=1)
+                acc = torch.sum(y == labels_hat).item() / (len(y) * 1.0)
+
+                # log the outputs separately for each dataloader
+                self.log_dict({f"val_loss_{dataloader_idx}": loss, "val_acc_{dataloader_idx}": acc})
```
Comment (bhimrazy) [inline src/lightning/pytorch/core/module.py]:
```suggestion
                self.log_dict({f"val_loss_{dataloader_idx}": loss, f"val_acc_{dataloader_idx}": acc})
```

File: src/lightning/pytorch/core/module.py [hunk 2]
```diff
@@ -875,7 +890,22 @@ def test_step(self, batch, batch_idx):
...
+                    loss = self.loss0(out, y)
+                else:
+                    loss = self.loss1(out, y)
+
+                # calculate acc
+                labels_hat = torch.argmax(out, dim=1)
+                acc = torch.sum(y == labels_hat).item() / (len(y) * 1.0)
+
+                # log the outputs separately for each dataloader
+                self.log_dict({f"test_loss_{dataloader_idx}": loss, "test_acc_{dataloader_idx}": acc})
```
Comment (bhimrazy) [inline src/lightning/pytorch/core/module.py]:
```suggestion
                self.log_dict({f"test_loss_{dataloader_idx}": loss, f"test_acc_{dataloader_idx}": acc})
```

**Review by bhimrazy — APPROVED** [review]

File: src/lightning/pytorch/core/module.py (hunk 1 above)
Comment (Borda) [reply src/lightning/pytorch/core/module.py]: good catch
